### DISCUSSION TITLE
Triggered reset question

### DISCUSSION THREAD

**Discussion by spaceg00se-r** [MiguelRipoll23/homebridge-securitysystem#516] [Q&A] [answered]
Hi

First of all I want to thank you for this awesome plugin. Keep up the great work! 

I have set up the security system successfully and it triggers when I open a door that is set up to do so (through the global trip switch and a corresponding automation). 
What I would now like to achieve is that I can create an automation that runs a "reset all alarm lights and sounds" scene if I set the status of the system to off while it is triggered. I hoped to do so through the optional sensor "Show Triggered Reset Sensor" which I thought would fire if the system is reset when I set the system to "off" (i also enabled the option "Reset using off mode"). However, the sensor is never firing if the system is triggered and I switch it to "off".
Do you have any suggestion on what I am doing wrong or do you see another way how I can run a scene to "clean up" all alarm activities when I switch the system to off while the system is triggered?

Thanks for your help!
Cheers, Chris

btw: there seems to be a little bug present in the "double-knock modes" configuration. The checkbox for "away" is always selected in the plugin settings although it is disabled in the configuration file.

**Comment by spaceg00se-r** [comment]
a small update... i found a workaround for my problem.

1. In homebridge i created a new virtual switch (stateful)
2. When the security system trip is triggered, i set the virtual switch to "on"
3. If I disarm the security system (set to "off") and the virtual switch is "on" - meaning it was triggered, I disable the stuff that was triggered through the alarm and I also set the virtual switch to "off". If the virtual switch is "off" it means that the system was not triggered and the security system just needs to be switched off.

Is there a more elegant way with the sensors mentioned in the initial question how I could achieve this behaviour?

**Comment by MiguelRipoll23** [comment, marked as answer]
Hey there,

Initially the reset sensor was only for the automated reset that happens when the security system is triggered for x amount of minutes. I agree that the reset sensor should also be triggered during a mode change while in triggered state.

Let’s convert this to an issue so this enhancement gets added to a new update. Will check the double knock bug you mentioned as well.

**Comment by spaceg00se-r** [comment]
Thanks for the super quick response. Awesome that you can look into it and maybe incorporate this in a future update. Already looking forward to it!

**Reply by MiguelRipoll23** [reply]
This is the issue you are talking about, right?

![image](https://github.com/MiguelRipoll23/homebridge-securitysystem/assets/3296866/8b7005f4-26dc-4885-b406-49bbc600e15b)

Unfortunately, there are are some limitations with the way plugin configuration is displayed in Homebridge so I can't automatically disable a checkbox if another related checkbox has been disabled. It can be disabled by the user though.

**Reply by spaceg00se-r** [reply]
I think my issue is slightly different. 

In the screenshot you can see the following. I manually disabled the "Home" mode, but I never selected the "Away" option in "Double-Knock Modes". Also if I go to the settings JSON file, I cannot see that double-knock is enabled for the "Away" mode.
  
![Screenshot 2024-02-03 at 21 50 25](https://github.com/MiguelRipoll23/homebridge-securitysystem/assets/145935806/c01c50b1-557c-4bbd-ad1a-bd0b6aac2508)

**Reply by MiguelRipoll23** [reply]
This will be fixed in the next update, issue found.

**Reply by spaceg00se-r** [reply]
Nice! Thanks!

**Comment by spaceg00se-r** [comment]
Did you already modify the plugin in the latest push to trigger the sensor, when the system is triggered? If so I assume that it is not yet in the official release, right?

**Reply by MiguelRipoll23** [reply]
Correct, it's the beta channel. It'll be released to the stable channel soon.

**Reply by spaceg00se-r** [reply]
Thanks a lot. Already updated and I'll try it in the coming days. A virtual beer for you! 🙌🏻 🍺
